data-prepper-plugins/http-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/http/service/HttpSinkService.java

```diff
@@ -144,28 +163,37 @@ public void output(Collection<Record<Event>> records) {
             this.currentBuffer = bufferFactory.getBuffer();
         }
         try {
+            OutputStream outputStream = currentBuffer.getOutputStream();
             records.forEach(record -> {
-                final Event event = record.getData();
                 try {
-                    currentBuffer.writeEvent(event.toJsonString().getBytes());
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
-                if (event.getEventHandle() != null) {
-                    this.bufferedEventHandles.add(event.getEventHandle());
-                }
-                if (ThresholdValidator.checkThresholdExceed(currentBuffer, maxEvents, maxBytes, maxCollectionDuration)) {
-                    final HttpEndPointResponse failedHttpEndPointResponses = pushToEndPoint(getCurrentBufferData(currentBuffer));
-                    if (failedHttpEndPointResponses != null) {
-                        logFailedData(failedHttpEndPointResponses, getCurrentBufferData(currentBuffer));
-                    } else {
-                        LOG.info("data pushed to the end point successfully");
+                    final Event event = record.getData();
+                    event.jsonBuilder().includeKeys(includeKeys);
```

Comment:
What are these two lines doing? I think you don't need them.

Reply:
Thanks for review @dlvenable , Sure, I shall remove it